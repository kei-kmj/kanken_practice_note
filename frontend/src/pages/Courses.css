.coursesTitle {
    position: relative;
    padding: 1rem 2rem;
}


