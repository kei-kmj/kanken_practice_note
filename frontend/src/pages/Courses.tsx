import * as React from 'react'
import { useRef, useState } from 'react'
import { useNavigate } from 'react-router-dom'
import { useFetchApi } from '../hooks/useFetchAPI'
import { Header} from '../components/shared/Header'
import { Logo } from '../components/shared/Logo'
import { LevelPanel } from '../components/courses/LevelPanel'
import { CategoryPanel } from '../components/courses/CategoryPanel'
import { QuizStartPanel } from '../components/courses/QuizStartPanel'
import { BackToTop } from '../components/shared/BackToTop'
import { Footer } from '../components/shared/Footer'
import { LOCAL_STORAGE_DATA } from '../hooks/useLocalStrage'
import { CATEGORY, LEVEL } from '../constants'
import '../App.css'

export const Courses:React.FC = () => {
  const { fetchRequest } = useFetchApi()
  const navigate = useNavigate()
  const [level, setLevel] = useState<string>('')
  const [category, setCategory] = useState<string>('')

  const ref = useRef<HTMLDivElement>(null!)
  const scrollToStartButton = () => {
    ref.current.scrollIntoView({
      behavior: "smooth",
    });
  }

  const selectCategory = (category: string) => {
    setCategory(category)
    scrollToStartButton()
  }

  const startQuiz = async (): Promise<void> => {
    const {quiz} = await fetchRequest(level, category, '10')
    // console.log(quiz)

    if (quiz.length === 0) {
      alert('出題できる問題がありません')
      return
    }
    navigate('/quiz', {state: {quiz}})
  }

  const repeatQuiz = () => {
    const quiz = []
    for (const key in localStorage) {
      if (!key.startsWith(LOCAL_STORAGE_DATA.KEY)) {
        continue
      } else if (level !== JSON.parse(localStorage.getItem(key) as string).level.toString()) {
        continue
      } else if (category === CATEGORY.ALL ||
        category === JSON.parse(localStorage.getItem(key) as string).category_id.toString()) {
        quiz.push(JSON.parse(localStorage.getItem(key) as string))
      }
    }
    console.log(quiz)
    if (quiz.length === 0) {
      alert('登録されている問題がありません')
      return
    }
    navigate('/quiz', {state: {quiz}})
  }
  return (<>
    <Header pageTitle="コース選択" description="問題の級と分野を選択します"/>
    <div className="flex flex-col min-h-screen">
      <main className="flex flex-col flex-1 top-0">
        <Logo/>
        <LevelPanel
          onClickSemiFirst={() => setLevel(LEVEL.SEMI_FIRST)}
          onClickFirst={() => setLevel(LEVEL.FIRST)}/>
        {level &&
        <div>
          <CategoryPanel
            onClickAll={() => selectCategory(CATEGORY.ALL)}
            onClickReading={() => selectCategory(CATEGORY.READING)}
            onClickExtraReading={() => selectCategory(CATEGORY.EXTRA_READING)}
            onClickDictation={() => selectCategory(CATEGORY.DICTATION)}
            onClickDictationIdiom={() => selectCategory(CATEGORY.DICTATION_IDIOM)}
            onClickMeaningIdiom={() => selectCategory(CATEGORY.MEANING_IDIOM)}
            onClickAntonymSynonym={() => selectCategory(CATEGORY.ANTONYM_SYNONYM)}
            onClickProverb={() => selectCategory(CATEGORY.PROVERB)}/>

          {category && <QuizStartPanel onClickStart={startQuiz} onClickRepeat={repeatQuiz}/>}
        </div>}
        <div ref={ref}/>
        <BackToTop/>
      </main>
      <Footer/>
    </div>
  </>)
}
