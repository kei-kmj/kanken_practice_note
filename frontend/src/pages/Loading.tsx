import * as React from 'react'

export const Loading:React.FC = () => {
  return (
    <div className="flex flex-col min-h-screen">
      <div className="flex justify-center">
        <div className="animate-spin h-10 w-10 border-4 border-accent rounded-full border-t-transparent"/><p className="text-accent text-xl">　...しばらくお待ちください</p>
      </div>
    </div>

  )
}
