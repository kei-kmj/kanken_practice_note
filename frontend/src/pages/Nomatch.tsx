import * as React from 'react'
import { BackToTop } from '../components/shared/BackToTop'

export const Nomatch:React.FC = () => {
  return (
    <div className="flex flex-col min-h-screen"><h2>404</h2>
      <h3>このページは存在しません</h3>
      <BackToTop/>
    </div>

  )
}
