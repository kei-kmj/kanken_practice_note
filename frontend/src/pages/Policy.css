.policy {
    background-color: #cc7000;
    height: 70px;
    text-align: center;
    color: #eaf4fc;
    padding: 0.5em 0.3em 0.8em 0.5em;
    border-radius: 8px;
}