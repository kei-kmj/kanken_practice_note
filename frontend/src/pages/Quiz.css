.modal {
    width: 380px;
}