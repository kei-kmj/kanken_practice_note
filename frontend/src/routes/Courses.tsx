import * as React from 'react'
import { useState } from 'react'
import { Link, useNavigate } from 'react-router-dom'
import Footer from '../components/Footer'
import { useFetchApi } from '../API'
import '../App.css'
import logo from '../../public/logo5.png'
import { LOCAL_STORAGE_DATA } from './Top'

const Courses = () => {
  const {fetchRequest} = useFetchApi()
  const navigate = useNavigate()
  const [level, setLevel] = useState('')
  const [category, setCategory] = useState('')

  const selectLevel = (level: string) => {
    setLevel(level)
  }

  const selectCategory = (category: string) => {
    setCategory(category)
  }

  const allQuizStart = async () => {
    const {quiz} = await fetchRequest(level, category, '10')
    console.log(quiz)

    if (level === '') {
      alert('級を選んでください')
      return
    } else if (category === '') {
      alert('分野を選んでください')
      return
    } else if (quiz.length === 0) {
      alert('出題できる問題がありません')
      return
    }
    navigate('/quiz', {state: {quiz}})
  }

  const repeatQuizStart = () => {
    if (level === '') {
      alert('級を選んでください')
      return
    } else if (category === '') {
      alert('分野を選んでください')
      return
    }
    const quiz = []
    for (let key in localStorage) {
      if (!key.startsWith(LOCAL_STORAGE_DATA.KEY)) {
        continue
      } else if (level !== JSON.parse(localStorage.getItem(key)).level.toString()) {
        continue
      } else if (category === '0' ||
        category === JSON.parse(localStorage.getItem(key)).category_id.toString()) {
        quiz.push(JSON.parse(localStorage.getItem(key)))
      }
    }
    console.log(quiz)
    if (quiz.length === 0) {
      alert('登録されている問題がありません')
      return
    }
    navigate('/quiz', {state: {quiz}})
  }
  return (<>
    <div>
      <main className="flex-grow">
        <div className="flex justify-left items-center m-2">
          {/* <h1 className="text-3xl">漢検練習帳</h1> */}
          {/* <img className="w-10" src={chara} alt={chara}/> */}
          <img className="w-40" src={logo} alt={logo}/>
        </div>

        <p className="text-xl flex justify-center mt-1">級を選んでください</p>
        <div className="flex justify-center m-0">
          <input id="radio1" className="radiobutton" name="level" hidden type="radio"
                 onClick={() => selectLevel('11')}
          />
          <label htmlFor="radio1">準1級</label>

          <input id="radio2" className="radiobutton" name="level" hidden type="radio" value="1"
                 onClick={() => selectLevel('1')}/>
          <label htmlFor="radio2">1 級</label>
        </div>

        <div>
          <div>
            <p className="text-xl flex justify-center mt-2">分野を選んでください</p>
            <div className="flex justify-center m-0">
              <input id="category1" className="radiobutton" name="category" hidden type="radio"
                     value="11"
                     onClick={() => selectCategory('0')}/>
              <label htmlFor="category1">全分野</label>

              <input id="category2" className="radiobutton" name="category" hidden type="radio"
                     value="3"
                     onClick={() => selectCategory('3')}/>
              <label htmlFor="category2">書 き</label>
            </div>
            <div className="flex justify-center">
              <input id="category3" className="radiobutton" name="category" hidden type="radio"
                     value="1"
                     onClick={() => selectCategory('1')}/>
              <label htmlFor="category3">読 み</label>

              <input id="category4" className="radiobutton" name="category" hidden type="radio"
                     value="2"
                     onClick={() => selectCategory('2')}/>
              <label htmlFor="category4">表外の読み</label>

            </div>
            <div className="flex justify-center">
              <input id="category5" className="radiobutton" name="category" hidden type="radio"
                     value="4"
                     onClick={() => selectCategory('4')}/>
              <label htmlFor="category5">四字熟語（書き）</label>

              <input id="category6" className="radiobutton" name="category" hidden type="radio"
                     value="5"
                     onClick={() => selectCategory('5')}/>
              <label htmlFor="category6">四字熟語（意味）</label>

            </div>
            <div className="flex justify-center">
              <input id="category7" className="radiobutton" name="category" hidden type="radio"
                     value="4"
                     onClick={() => selectCategory('6')}/>
              <label htmlFor="category7">対義語・類義</label>

              <input id="category8" className="radiobutton" name="category" hidden type="radio"
                     value="5"
                     onClick={() => selectCategory('7')}/>
              <label htmlFor="category8">故事成語・諺</label>
            </div>

            <p className="text-xl flex justify-center mt-5">新規又は復習を選んでください</p>
            <div className="flex justify-center">
              <div>
                <button className="btn btn-wide m-2 btn-primary b" onClick={allQuizStart}>新 規
                </button>
                <p className="flex justify-center">選んだ分野のすべての問題から出題されます</p>
              </div>
              <div>
                <button className="btn btn-wide m-2 btn-primary b" onClick={repeatQuizStart}>復 習
                </button>
                <p className="flex justify-center">「復習する」にチェックした問題から出題されます</p>
              </div>
            </div>
            <div className="grid grid-flow-col gap-4 link flex justify-end m-5">
              <Link to="/vite-dev">トップ画面に戻る</Link>
            </div>
          </div>
        </div>
      </main>

      <Footer/>

    </div>
  </>)
}
export default Courses
