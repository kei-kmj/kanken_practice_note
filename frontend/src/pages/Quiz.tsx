import * as React from 'react'
import { useState } from 'react'
import { useNavigate, useLocation } from 'react-router-dom'
import { LOCAL_STORAGE_DATA, saveJSON, removeJSON } from '../hooks/useLocalStrage'
import { Header } from '../components/shared/Header'
import { Logo } from '../components/shared/Logo'
import { Question } from '../components/quiz/Question'
import { Alternatives } from '../components/quiz/Alternatives'
import { DisplayCorrect } from '../components/quiz/DisplayCorrect'
import { DisplayIncorrect } from '../components/quiz/DisplayIncorrect'
import { Note } from '../components/quiz/Note'
import { RepeatButton } from '../components/quiz/RepeatButton'
import { NextQuiz } from '../components/quiz/NextQuiz'
import { AbortConfirm } from '../components/quiz/AbortConfirm'
import { EndOfQuiz } from '../components/quiz/EndOfQuiz'
import { Score } from '../components/quiz/Score'
import { Footer } from '../components/shared/Footer'
import { QuestionType } from '../types/quizTypes'
import './Quiz.css'

export const Quiz:React.FC = () => {
  const navigate = useNavigate()
  const location = useLocation()

  const data = location.state as { quiz:QuestionType }
  // console.log(data)
  const [currentQuiz, setCurrentQuiz] = useState<number>(0)
  const [score, setScore] = useState<number>(0)

  const correctAnswer = data.quiz[currentQuiz].answers.filter((answer: { correctness: boolean }) => {
    if (answer.correctness) {
      return answer
    } else return null
  })

  const finishQuiz = () => {
    navigate('/courses')
  }

  const addScore = (answer: { correctness: boolean }) => {
    if (answer.correctness) {
      setScore(score + 1)
    }
  }

  const handleToggle = (e: { target: { checked: boolean } }) => {
    if (e.target.checked) {
      saveJSON(`${LOCAL_STORAGE_DATA.KEY}${data.quiz[currentQuiz].id}`, data.quiz[currentQuiz])
    } else {
      removeJSON(`${LOCAL_STORAGE_DATA.KEY}${data.quiz[currentQuiz].id}`)
    }
  }

  return (<>
    <Header pageTitle="問題です" description="問題を出題します" title="漢検練習帳"/>
    <div className="flex flex-col min-h-screen">
      <div className="flex justify-left">
        <Logo />
      </div>
      <main className="flex-grow">
        <Question data={data} currentQuiz={currentQuiz}/>
        {data.quiz[currentQuiz].answers.map((answer: { id: string; correctness: boolean; answer: string }) =>
          <div className="flex justify-center mt-2" key={answer.id}>

            <Alternatives answer={answer} onClick={() => addScore(answer)}/>

            <input type="checkbox" id={answer.id} className="modal-toggle"/>
            <div className="modal">
              <div className="modal-box bg-white">
                <div>
                  <div className="flex justify-center">
                    {answer.correctness.toString() === 'true' ? <DisplayCorrect/> : <DisplayIncorrect/>}
                  </div>
                  <p className="text-2xl flex justify-center m-5 ">答え：{correctAnswer[0].answer}</p>

                  <Note data={data} currentQuiz={currentQuiz}/>

                  <RepeatButton data={data} currentQuiz={currentQuiz} onClick={(e) => handleToggle(e)}/>

                  {currentQuiz === data.quiz.length - 1 ? <Score score={score} data={data}/> : null}

                  {currentQuiz < data.quiz.length - 1
                    ? <NextQuiz answer={answer} onClick={() => {
                      setCurrentQuiz(currentQuiz + 1)
                    }}/>
                    : <EndOfQuiz answer={answer} onClick={finishQuiz}/>}

                  {currentQuiz < data.quiz.length - 1
                    ? <AbortConfirm answer={answer} onClick={finishQuiz}/>
                    : null}
                </div>
              </div>
            </div>
          </div>)}
      </main>
      <Footer/>
    </div>
  </>)
}