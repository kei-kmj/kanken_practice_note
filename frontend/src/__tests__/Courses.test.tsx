import { describe, test } from 'vitest'
import { BrowserRouter as Router } from 'react-router-dom';
import { render, screen } from '@testing-library/react'
import { Courses } from '../pages/Courses'
import userEvent from '@testing-library/user-event'

describe('Coursesページレンダリング', () => {
  test('スナップショット一致', () => {
    const {container} = render(<Router><Courses/></Router>)
    expect(container.firstChild).toMatchSnapshot()
  })

  test('すべての要素が正常にレンダリングされている', () => {
    render(<Router><Courses/></Router>)
    expect(screen.getAllByRole("img")).toBeTruthy()

    expect(screen.getByText("級を選んでください")).toBeInTheDocument()
    expect(screen.getByLabelText("準1級")).toBeInTheDocument()
    expect(screen.getByLabelText("1 級")).toBeInTheDocument()

    expect(screen.getAllByRole("link")).toBeTruthy()
    expect(screen.getAllByRole("generic")).toBeTruthy()

    expect(screen.getByText("「漢検」「漢字検定」は")).toBeInTheDocument()
    expect(screen.getByText("公益財団法人 日本漢字能力検定協会の登録商標です")).toBeInTheDocument()
  })
})
