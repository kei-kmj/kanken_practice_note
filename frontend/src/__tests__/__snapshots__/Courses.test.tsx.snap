// Vitest Snapshot v1

exports[`Coursesページレンダリング > スナップショット一致 1`] = `
<div
  class="flex flex-col min-h-screen"
>
  <main
    class="flex flex-col flex-1 top-0"
  >
    <div
      class="flex justify-left items-center m-2"
    >
      <img
        alt="/vite-test/public/logo.png"
        class="w-40"
        src="/vite-test/public/logo.png"
      />
    </div>
    <p
      class="text-xl flex justify-center mt-1"
    >
      級を選んでください
    </p>
    <div
      class="flex justify-center m-0"
    >
      <input
        class="radiobutton"
        hidden=""
        id="radio1"
        name="level"
        role="radio"
        type="radio"
      />
      <label
        for="radio1"
      >
        <span
          class="dummy-radio"
        >
          <span
            class="dummy-radio-check"
          />
        </span>
         
        準1級
      </label>
    </div>
    <div
      class="flex justify-center m-0"
    >
      <input
        class="radiobutton"
        disabled=""
        hidden=""
        id="radio2"
        name="level"
        role="radio"
        type="radio"
      />
      <label
        for="radio2"
      >
        <span
          class="dummy-radio"
        >
          <span
            class="dummy-radio-check"
          />
        </span>
         
        1 級
      </label>
    </div>
    <div />
    <div
      class="grid grid-flow-col gap-4 link flex justify-end m-5"
    >
      <a
        href="/"
      >
        トップ画面に戻る
      </a>
    </div>
  </main>
  <footer
    class="margin-top-auto p-3 bg-accent text-sm text-pink-100"
  >
    <p
      class="flex justify-center text-xs"
    >
      「漢検」「漢字検定」は
    </p>
    <p
      class="flex justify-center text-xs"
    >
       公益財団法人 日本漢字能力検定協会の登録商標です
    </p>
    <div
      class="flex justify-center mt-1 grid grid-flow-col gap-4"
    >
      <a
        href="/policy"
      >
        <span
          class="link"
        >
          アプリケーションポリシー
        </span>
      </a>
      <a
        href="/support"
      >
        <span
          class="link"
        >
          お問い合わせ
        </span>
      </a>
      <span
        class="text-xl"
      >
        <a
          href="https://twitter.com/home"
          rel="noopener noreferrer"
          target="_blank"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 512 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M459.37 151.716c.325 4.548.325 9.097.325 13.645 0 138.72-105.583 298.558-298.558 298.558-59.452 0-114.68-17.219-161.137-47.106 8.447.974 16.568 1.299 25.34 1.299 49.055 0 94.213-16.568 130.274-44.832-46.132-.975-84.792-31.188-98.112-72.772 6.498.974 12.995 1.624 19.818 1.624 9.421 0 18.843-1.3 27.614-3.573-48.081-9.747-84.143-51.98-84.143-102.985v-1.299c13.969 7.797 30.214 12.67 47.431 13.319-28.264-18.843-46.781-51.005-46.781-87.391 0-19.492 5.197-37.36 14.294-52.954 51.655 63.675 129.3 105.258 216.365 109.807-1.624-7.797-2.599-15.918-2.599-24.04 0-57.828 46.782-104.934 104.934-104.934 30.213 0 57.502 12.67 76.67 33.137 23.715-4.548 46.456-13.32 66.599-25.34-7.798 24.366-24.366 44.833-46.132 57.827 21.117-2.273 41.584-8.122 60.426-16.243-14.292 20.791-32.161 39.308-52.628 54.253z"
            />
          </svg>
        </a>
      </span>
    </div>
    <div
      class="flex justify-center"
    >
      <p
        class="flex justify-center m-1 text-xs"
      >
        ©kei-kmj 
        2023
         
      </p>
      <p
        class="flex justify-center m-1 text-xs"
      >
        <a
          href="https://github.com/kei-kmj/kanken_practice_note"
          rel="noopener noreferrer"
          target="_blank"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 496 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M165.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6zm-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3zm44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9zM244.8 8C106.1 8 0 113.3 0 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C428.2 457.8 496 362.9 496 252 496 113.3 383.5 8 244.8 8zM97.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1zm-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7zm32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1zm-11.4-14.7c-1.6 1-1.6 3.6 0 5.9 1.6 2.3 4.3 3.3 5.6 2.3 1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2z"
            />
          </svg>
        </a>
      </p>
      <p
        class="flex justify-center m-1 text-xs"
      >
        <a
          href="https://twitter.com/kei16180339"
          rel="noopener noreferrer"
          target="_blank"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 512 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M459.37 151.716c.325 4.548.325 9.097.325 13.645 0 138.72-105.583 298.558-298.558 298.558-59.452 0-114.68-17.219-161.137-47.106 8.447.974 16.568 1.299 25.34 1.299 49.055 0 94.213-16.568 130.274-44.832-46.132-.975-84.792-31.188-98.112-72.772 6.498.974 12.995 1.624 19.818 1.624 9.421 0 18.843-1.3 27.614-3.573-48.081-9.747-84.143-51.98-84.143-102.985v-1.299c13.969 7.797 30.214 12.67 47.431 13.319-28.264-18.843-46.781-51.005-46.781-87.391 0-19.492 5.197-37.36 14.294-52.954 51.655 63.675 129.3 105.258 216.365 109.807-1.624-7.797-2.599-15.918-2.599-24.04 0-57.828 46.782-104.934 104.934-104.934 30.213 0 57.502 12.67 76.67 33.137 23.715-4.548 46.456-13.32 66.599-25.34-7.798 24.366-24.366 44.833-46.132 57.827 21.117-2.273 41.584-8.122 60.426-16.243-14.292 20.791-32.161 39.308-52.628 54.253z"
            />
          </svg>
        </a>
      </p>
    </div>
  </footer>
</div>
`;
